code/datums/spells/lichdom.dm

```diff
@@ -81,6 +76,11 @@
 			current_body = lich
 			lich.Weaken(10+10*resurrections)
 			++resurrections
+			lich.equip_to_slot_or_del(new /obj/item/clothing/shoes/sandal(lich), slot_shoes)
+			lich.equip_to_slot_or_del(new /obj/item/clothing/under/color/black(lich), slot_w_uniform)
+			lich.equip_to_slot_or_del(new /obj/item/clothing/suit/wizrobe/black(lich), slot_wear_suit)
+			lich.equip_to_slot_or_del(new /obj/item/clothing/head/wizard/black(lich), slot_head)
+
```

Comment:
The reason this change fixes the behavior, at least from what I know of the code, is that changing the species causes the limbs to be regenerated - and this causes the code that makes equipment drop upon limb removal run, causing the hat and sandals to fall off

Reply:
I looked into this - and can confirm it. Moving set_species("Skeleton") below the equipping line  caused the undesired effect of dropping hat and feet.